shell/src/state/streaming_state.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::collections::HashMap;
+
+use futures::task::Waker;
+use uuid::Uuid;
+
+pub trait StreamCounter {
```

Comment:
@adonagy 
I would better move this to the `shell-integration` module, because in the end, I want to remove `shell` dependency from `rpc/Cargo.toml`

Reply:
Ok, will do